pkg/bgpv1/manager/route_policy_reconciler.go

```diff
@@ -168,8 +176,25 @@ func (r *RoutePolicyReconciler) pathAttributesToPolicy(attrs v2alpha1api.CiliumB
 	}
 
 	switch attrs.SelectorType {
+	case v2alpha1api.CiliumPodIPPoolSelectorName:
+		localPools := r.populateLocalPools(params.CiliumNode)
+		for _, pool := range r.podStore.List() {
+			if attrs.Selector != nil && !labelSelector.Matches(labels.Set(pool.Labels)) {
+				continue
+			}
+			// only advertise pool cidrs that have been allocated to the local node.
+			if cidrs, ok := localPools[pool.Name]; ok {
+				for _, cidr := range cidrs {
+					if cidr.Addr().Is4() {
+						v4Prefixes = append(v4Prefixes, &types.RoutePolicyPrefixMatch{CIDR: cidr, PrefixLenMin: maxPrefixLenIPv4, PrefixLenMax: maxPrefixLenIPv4})
```

Comment:
`PrefixLenMin` and `PrefixLenMax` should be equal to the actual Pool prefix length. This is most likely the reason why the policy does not apply on the advertisement.

Reply:
@rastislavs thanks for the guidance. ^ was indeed the issue. Commit b9cf743 resolves this conversation.